### DISCUSSION TITLE
database modeling

### DISCUSSION THREAD

**Discussion by jadsongmatos** [mermaid-js/mermaid#4267] [Ideas]
BR Modelo Web is an Open Source project for entity-relationship database modeling. BR Modelo Web allows you to create logical and conceptual models, including the definition of entities, attributes and relationships, as well as the use of constraints to ensure data integrity.

It would be interesting to have a textual way to create diagrams of conceptual models that are simpler to generate entity-relationship diagrams.

[https://github.com/brmodeloweb/brmodelo-app](https://github.com/brmodeloweb/brmodelo-app)

**Comment by JesseRigon** [comment]
Entity Relationship Diagrams are standard in Mermaidjs and have pretty much the features of a database modeler except export to SQL syntax for database generation, imports and stuff like that. 

Another problem with using mermaidjs for database modeling is that the relationships are only available at the entity level and not at the attribute or method level which is really limiting.

**Comment by vinicius507** [comment]
Yes, I'd love this. Is this something the maintainers would consider accepting a PR for?

**Comment by EmnaBarbouch** [comment]
Great idea! Attribute-level relationships and SQL export would make Mermaid's `erDiagram` much more useful for real database modeling workflows. Would love to see this move forward as a PR.
